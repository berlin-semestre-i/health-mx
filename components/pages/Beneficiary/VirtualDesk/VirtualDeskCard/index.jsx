import React from 'react'
import { Card, Icon } from 'semantic-ui-react'
import styled from 'styled-components'
import media from 'styled-media-query'

const VirtualDeskCard = ({ children, header, iconName, iconColor }) => (
  <DeskCard>
    <Card.Content>
      <Card.Header>
        { header }
        <AddIcon>
          <Icon color="blue" name="plus" />
        </AddIcon>
        <Secondary>
          {iconName && iconColor && (
            <Icon color={iconColor} name={iconName} />
          )}
          {iconName && !iconColor && (
            <Icon name={iconName} />
          )}
        </Secondary>
      </Card.Header>
      { children }
    </Card.Content>
  </DeskCard>
)

export default VirtualDeskCard

const DeskCard = styled(Card)`
  && {
    width: 100%;
  }

  ${media.lessThan('medium')`
    &&.nth-col-first {
      margin-top: 1em;
    }
  `}

  &&.loaderCard .ui.loader:after {
    border-color: rgba(0,0,0,0.2) transparent transparent;
  }
`

const AddIcon = styled.div`
    float: right;
    cursor: pointer;
    color: #3d5170;
    text-transform: none;
    letter-spacing: 0.5px;
    margin-left: 1rem;
    font-size: 18px;

    ${media.lessThan('small')`
        >span {
        display: none;
         }
    `}
`

const Secondary = styled.div`
    float: right;
    cursor: pointer;
    color: #3d5170;
    text-transform: none;
    letter-spacing: 0.5px;
    font-size: 18px;

    ${media.lessThan('small')`
        >span {
        display: none;
         }
    `}
`