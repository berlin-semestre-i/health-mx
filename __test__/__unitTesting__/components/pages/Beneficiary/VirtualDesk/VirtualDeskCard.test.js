import React from 'react'
import { shallow, mount } from 'enzyme'
// eslint-disable-next-line max-len
import VirtualDeskCard from '../../../../../../components/pages/Beneficiary/VirtualDesk/VirtualDeskCard/index'
import { MockedProvider } from 'react-apollo/test-utils'
import {
  graphQLMocks,
  graphQLErrorMocks,
} from '../../../../__mocks__/requests/articles'

describe('Default table', () => {
  it('renders', () => {
    const component = shallow(<VirtualDeskCard />)
    expect(component).toMatchSnapshot()
  })

  it('has add icon', () => {
    const component = mount(
      <MockedProvider mocks={graphQLMocks} >
        <VirtualDeskCard />
      </MockedProvider>
    )
    expect(component.find('i.plus').length).toBe(1)
    component.unmount()
  })

  it('has move icon', () => {
    const component = mount(
      <MockedProvider mocks={graphQLMocks} >
        <VirtualDeskCard iconName="move"/>
      </MockedProvider>
    )
    expect(component.find('i.move').length).toBe(1)
    component.unmount()
  })
})

