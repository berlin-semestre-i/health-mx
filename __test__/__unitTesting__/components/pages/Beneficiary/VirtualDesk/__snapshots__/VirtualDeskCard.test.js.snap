// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default table renders 1`] = `
<VirtualDeskCard__DeskCard>
  <CardContent>
    <CardHeader>
      <VirtualDeskCard__AddIcon>
        <Icon
          as="i"
          color="blue"
          name="plus"
        />
      </VirtualDeskCard__AddIcon>
      <VirtualDeskCard__Secondary />
    </CardHeader>
  </CardContent>
</VirtualDeskCard__DeskCard>
`;
